### PR TITLE
[DOCS] Reformat `parent_id` query docs

### PR DESCRIPTION
**Pull request by jrodewig**
Rewrites the `parent_id` query to use the new query format.

This creates separate sections for example requests, parameters, and notes.

This is part of #40977, an effort to standardize documentation for query types.

## Before
<details>
 <summary>Before image</summary>
Parent ID Query - Before
</details>

## After
<details>
 <summary>After image</summary>
Parent ID Query - After
</details>

### CODE REVIEW

File: docs/reference/query-dsl/parent-id-query.asciidoc [hunk 1]
```diff
@@ -1,68 +1,113 @@
...
+[[parent-id-query-ex-request]]
+==== Example request
+
+[[parent-id-index-setup]]
+===== Index setup
+To use the `parent_id` query, your index must include a <<parent-join,join>>
+field mapping. To see how you can set up an index for the `parent_id` query, try
+the following example.
+
+. Create an index with a <<parent-join,join>> field mapping.
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/parent-id-query.asciidoc]: `.` dot at the beginning of the sentence?
Here and in a couple of places below.

File: docs/reference/query-dsl/parent-id-query.asciidoc [hunk 2]
```diff
@@ -1,68 +1,113 @@
...
+===== Example query
+
+The following search returns child documents for a parent document with an ID of
+`1`.
 
 [source,js]
---------------------------------------------------
-GET /my_index/_search
+----
+GET /my-index/_search
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/parent-id-query.asciidoc]: should we have a unified format in all examples:  either put `/` before index name or not? `/my-index/_search` -> `my-index/_search`

**Review by mayya-sharipova — APPROVED** [review]

File: docs/reference/query-dsl/parent-id-query.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/query-dsl/parent-id-query.asciidoc]: Thank you for catching this. I added leading slashes with 647110c.

File: docs/reference/query-dsl/parent-id-query.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/query-dsl/parent-id-query.asciidoc]: This renders as an ordered list in the HTML output. Here's a screenshot:

Screen Shot 2019-07-19 at 10 35 31 AM

**Comment by jrodewig** [comment]
Thanks again for your review @mayya-sharipova!

**Pull request merged**